warrior/Framework/ClassUtils/netconf_utils_class.py

```diff
@@ -66,6 +66,9 @@ def open(self, session_kwds):
                                               nc_session['password'],
                                               nc_session['hostkey_verify'],
                                               nc_session['protocol_version'])
+            # If connection fails return False
+            if self.nc_manager is None:
```

Comment:
Hi @venkat1077 
I believe using status = self.nc_manager.isOpen here
and let the except block catch the exception and return false will be a cleaner approach
In the case where netconf.connect changes to return False, this approach may work better

Reply:
I agree @kaheichan, thanks for the suggestion.
Reasons for keeping this way (i) Most of the known exceptions(during socket/transport creation) are handled and the error_messages and/or tracebacks are printed inside netconf.connect class_method(class_name:client). Since netconf.connect function returns 'None' only when any of these known exceptions occur, I decided not to raise the exception one more time. (ii) To maintain the format of the return statement(last), common across all the methods of netconf_utils_class.py.